frontend/src/pages/admin/access/ManageAccessRequests.tsx

```diff
@@ -0,0 +1,138 @@
+import * as API from 'constants/API';
+import { IAccessRequest } from 'interfaces';
+import _ from 'lodash';
+import React, { useEffect, useState } from 'react';
+import { Button, Col, Container, Row, Spinner } from 'react-bootstrap';
+import { useDispatch, useSelector } from 'react-redux';
+import { IAccessRequestState, IColumnFilter, ISort } from 'reducers/accessRequestReducer';
+import { RootState } from 'reducers/rootReducer';
+import { IPaginate, toApiPaginateParams } from 'utils/CommonFunctions';
+import { columnDefinitions } from './constants/constants';
+import { AccessRequestActions } from './components/AccessRequestActions';
+
+import './ManageAccessRequests.scss';
+import 'ag-grid-community/dist/styles/ag-grid.css';
+import 'ag-grid-community/dist/styles/ag-theme-balham.css';
+
+const ManageAccessRequests = () => {
+  const dispatch = useDispatch();
+  const MAX_ACCESS_RESULTS_PER_PAGE = 5;
+  const pagedAccessRequests = useSelector<RootState, IPaginate>(
+    state => (state.accessRequest as IAccessRequestState)?.pagedAccessRequests,
+  );
+
+  const gridFilterModel = useSelector<RootState, { [key: number]: IColumnFilter }>(
+    state => (state.accessRequest as IAccessRequestState).filters || {},
+  );
+
+  const gridSortModel = useSelector<RootState, ISort[]>(
+    state => (state.accessRequest as IAccessRequestState).sorting || [],
+  );
+
+  const gridRowSelections = useSelector<RootState, string[]>(
+    state => (state.accessRequest as IAccessRequestState).selections || [],
+  );
+
+  const [selectedRequests, setSelectedRequests] = useState<IAccessRequest[]>([]);
+
+  const [gridApi, setGridApi] = useState<GridApi | null>(null);
+
+  const requestAccess = useSelector<RootState, IGenericNetworkAction>(
+    state => (state.network as any)[actionTypes.GET_REQUEST_ACCESS] as IGenericNetworkAction,
+  );
+
+  const updateRequestAccessAdmin = useSelector<RootState, IGenericNetworkAction>(
+    state =>
+      (state.network as any)[actionTypes.UPDATE_REQUEST_ACCESS_ADMIN] as IGenericNetworkAction,
+  );
+
+  useEffect(() => {
+    if (!updateRequestAccessAdmin?.isFetching) {
+      const paginateParams: API.IPaginateAccessRequests = toApiPaginateParams(
+        0,
+        MAX_ACCESS_RESULTS_PER_PAGE,
+      );
+      paginateParams.status = 'ON_HOLD';
+      dispatch(getAccessRequestsAction(paginateParams));
+    }
+  }, [updateRequestAccessAdmin, dispatch]);
+
+  useEffect(() => {
+    if (gridApi) {
+      gridApi.setFilterModel(gridFilterModel || {});
+      gridApi.setSortModel(gridSortModel || []);
+      gridRowSelections.forEach(x => {
+        gridApi.getRowNode(x)?.setSelectedInitialValue(true);
+      });
+    }
+  }, [gridApi]);
+
+  const updateSelectionList = () => {
+    setSelectedRequests(gridApi?.getSelectedNodes().map(x => x.data) || []);
+    dispatch(getAccessRequestsSelectAction(gridApi?.getSelectedNodes().map(x => x.id) || []));
+  };
+
+  const setFilterModel = () => {
```

Comment:
Is there an advantage to storing the filter/sort/selections in redux instead of local component state?

My first thought is that it would allow external components to modify the filter/sort/selections of this form directly, is that something that we want to support?

I don't see a major disadvantage to doing this besides performance, so I don't mind keeping this in even if these values will only ever be tracked by this component.

Reply:
When the component re-renders after updating the status of an access request, the table loses its state. I moved the table state to redux to maintain the table state when re-rendering